source/Applications/Advanced/TransformPointCloudViaArucoMarker/TransformPointCloudViaArucoMarker.cpp

```diff
@@ -0,0 +1,313 @@
+                                        + (x - xRoundedDown) / (xRoundedUp - xRoundedDown) * q21.z };
+            // f(x,y2) = (x2 - x)/(x2 - x1) * f(q12) + (x - x1)/(x2 - x1) * f(q22)
+            const cv::Point3f fxy2{ (xRoundedUp - x) / (xRoundedUp - xRoundedDown) * q12.x
+                                        + (x - xRoundedDown) / (xRoundedUp - xRoundedDown) * q22.x,
+                                    (xRoundedUp - x) / (xRoundedUp - xRoundedDown) * q12.y
+                                        + (x - xRoundedDown) / (xRoundedUp - xRoundedDown) * q22.y,
+                                    (xRoundedUp - x) / (xRoundedUp - xRoundedDown) * q12.z
+                                        + (x - xRoundedDown) / (xRoundedUp - xRoundedDown) * q22.z };
+
+            // Linear interpolation in y direction
+            // f(x,y) = (y2 - y)/(y2 - y1) * f(x,y1) + (y - y1)/(y2 - y1) * f(x,y2)
+            markerPoints3D.emplace_back(cv::Point3f{ (yRoundedUp - y) / (yRoundedUp - yRoundedDown) * fxy1.x
+                                                         + (y - yRoundedDown) / (yRoundedUp - yRoundedDown) * fxy2.x,
+                                                     (yRoundedUp - y) / (yRoundedUp - yRoundedDown) * fxy1.y
+                                                         + (y - yRoundedDown) / (yRoundedUp - yRoundedDown) * fxy2.y,
+                                                     (yRoundedUp - y) / (yRoundedUp - yRoundedDown) * fxy1.z
+                                                         + (y - yRoundedDown) / (yRoundedUp - yRoundedDown) * fxy2.z });
+        }
+
+        return markerPoints3D;
+    }
+
+    Zivid::Matrix4x4 transformationMatrix(const cv::Matx33d &rotationMatrix, const cv::Vec3d &translationVector)
+    {
+        auto transformMatrix = Zivid::Matrix4x4{ 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1 };
+
+        for(size_t row = 0; row < rotationMatrix.rows; row++)
+        {
+            for(size_t col = 0; col < rotationMatrix.cols; col++)
+            {
+                transformMatrix(row, col) = rotationMatrix(row, col);
+            }
+        }
+
+        transformMatrix(0, 3) = translationVector[0];
+        transformMatrix(1, 3) = translationVector[1];
+        transformMatrix(2, 3) = translationVector[2];
+
+        return transformMatrix;
+    }
+
+    Zivid::Matrix4x4 estimateArUcoMarkerPose(const Zivid::PointCloud &pointCloud,
+                                                  const int &markerId,
+                                                  std::vector<cv::Point2f> &markerCorners)
+    {
+        // Extracting 2D corners and estimateing 2D center
+
+        const auto center2D = estimate2DMarkerCenter(markerId, markerCorners);
+
+        // Estimating 3D corners and center from 2D data
+        const auto corners3D = estimate3DMarkerPoints(pointCloud, markerCorners);
+        const auto center3D = estimate3DMarkerPoints(pointCloud, std::vector<cv::Point2f>(1, center2D));
```

Comment:
You can write this nicer as:
```
const auto center3D = estimate3DMarkerPoints(pointCloud, { center2D });
```

Reply:
Maybe even save the one element instead of the vector:
```
const auto center3D = estimate3DMarkerPoints(pointCloud, { center2D })[0];
```
since your left hand name is singular. Then you can remove the "[0]" from the origin calculation line.

Reply:
I think the second option (removing [0] from the origin line) would not work because I use the same function for the vector of points (4 x 2) in the above line.

Going for the first option.